enhancements/rhcos/static-networking-enhancements.md

```diff
@@ -0,0 +1,104 @@
+---
+title: improved-static-networking-configuration
+authors:
+  - "@miabbott"
+reviewers:
+approvers:
+  - "@ashcrow"
+  - "@cgwalters"
+  - "@crawford"
+  - "@imcleod"
+  - "@runcom"
+creation-date: 2020-04-22
+last-updated: 2020-04-22
+status: provisional
+see-also: https://github.com/openshift/enhancements/pull/210
+replaces:
+superseded-by:
+---
+
+# Improved Static Networking Configuration
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+This enhancement describes improvements to how static networking is configured on Fedora CoreOS (FCOS) and RHEL CoreOS (RHCOS).  It covers improving the configuration of static networking for bare metal and VMWare hosts.
+
+## Motivation
+Many customer environments require static networking and do not allow any type of DHCP server. This is most prominent in VMware environments where they struggle with the OVA images and are resisting the bare metal installer (via ISO) approach. This is also a challenge in bare metal environments due to an unfriendly and time sensitive interactive flow (i.e. catching the `grub` prompt). Due to the intersections of Ignition, Platform ID, and active networking in the `initrd`, many of the current RHCOS images append `ip=dhcp,dhcp6` and catching the bootloader on the console is not a pleasant experience. (Even if a static `ip=::::` is passed it will fail as the platform ID will insert `ip=dhcp` and conflict)
```

Comment:
> (Even if a static `ip=::::` is passed it will fail as the platform ID will insert `ip=dhcp` and conflict)

It might be best to just remove that comment. I don't think it is exactly true because at the point the user stops grub and hand edits the kernel command line the `ip=dhcp,dhcp6` should be there on the command line so they can delete it and then add their own.

Reply:
@miabbott thoughts on this change?

Reply:
I have it pending in my local repo...will have an update soon.